src/notification/link_poll_event.ts

```diff
@@ -0,0 +1,23 @@
+import {Link, Store} from '../store/model';
+
+type RequestFailed = {
+  failureReason: 'request_failed';
+  statusCode: number;
+};
+type Captcha = {failureReason: 'captcha'};
+type MaxPriceExceeded = {failureReason: 'max_price'; maxPrice: number};
+type OutOfStock = {failureReason: 'out_of_stock'};
+type BannedSeller = {failureReason: 'banned_seller'};
+type LinkPollError = {result: 'failure'} & (
+  | RequestFailed
+  | Captcha
+  | MaxPriceExceeded
+  | OutOfStock
+  | BannedSeller
+);
+type LinkPollSuccess = {result: 'in_stock'; url: string};
+
+export type LinkPollEvent = (LinkPollError | LinkPollSuccess) & {
+  link: Link;
+  store: Store;
+};
```

Comment:
The `LinkPollEvent` could be made shorter but I liked to express it in a way that prevents representing illegal states.